File: Project_Java/src/PositionRule.java
public class PositionRule {
    Rule rule;
    int position;

    PositionRule(Rule r, int p){
        rule=r;
        position=p;
    }

    String getNextSymbol() {
        return rule.symbols.get(position);
    }

    boolean isLast() {
        return position == rule.symbols.size();
    }
}

File: Project_Python/src/PositionRule.py
class PositionRule:
    def __init__(self, r, p):
        self.rule = r
        self.position = p

    def getNextSymbol(self):
        return self.rule.symbols[self.position]

    def isLast(self):
        return self.position == len(self.rule.symbols)
